marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/smart-mastering/entity-driven/test-data/householdAction.sjs

```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const sem = require("/MarkLogic/semantics.xqy");
+const householdPredicateIRI = sem.iri('http://marklogic.com/datahub/test/household#memberOf');
+
+function householdAction(uri, matches, options) {
+    declareUpdate();
+    const allURIs = [uri].concat(matches.map((match) => match.uri)).sort();
+    const householdIRI = sem.iri(`http://marklogic.com/datahub/test/household/${xdmp.md5(allURIs)}`);
+    sem.rdfInsert(allURIs.map((uri) => sem.triple(sem.iri(uri), householdPredicateIRI, householdIRI)));
```

Comment:
Is inserting data via a match action something we'd encourage? I ask because I'm wondering if we can eventually shift to a model where everything to be written gets stored in-memory first and then is written after all step processing has finished.